src/extensions/logic.rs

```diff
@@ -29,65 +29,67 @@ pub fn resource() -> Resource {
     const H_INT: TypeParam = TypeParam::Value(HashableType::Int(8));
     let mut resource = Resource::new(resource_id());
 
-    let not_op = OpDef::new_with_custom_sig(
-        "Not".into(),
-        "logical 'not'".into(),
-        vec![],
-        HashMap::default(),
-        |_arg_values: &[TypeArg]| {
-            Ok((
-                vec![bool_type()].into(),
-                vec![bool_type()].into(),
-                ResourceSet::default(),
-            ))
-        },
-    );
-
-    let and_op = OpDef::new_with_custom_sig(
-        "And".into(),
-        "logical 'and'".into(),
-        vec![H_INT],
-        HashMap::default(),
-        |arg_values: &[TypeArg]| {
-            let a = arg_values.iter().exactly_one().unwrap();
-            let n: u128 = match a {
-                TypeArg::Int(n) => *n,
-                _ => {
-                    return Err(TypeArgError::TypeMismatch(a.clone(), H_INT).into());
-                }
-            };
-            Ok((
-                vec![bool_type(); n as usize].into(),
-                vec![bool_type()].into(),
-                ResourceSet::default(),
-            ))
-        },
-    );
+    resource
+        .add_op_custom_sig(
```

Comment:
Could we just use `add_op()` for this?

Reply:
the `SignatureFunc` enum is private (which makes sense to me, given the serialisation trickery used internally)